src/core/overflowsafe_type.hpp

```diff
@@ -39,25 +39,41 @@ class OverflowSafeInt
 	/**
 	 * Safe implementation of addition.
 	 * @param other the amount to add
-	 * @note when the addition would yield more than T_MAX (or less than T_MIN),
-	 *       it will be T_MAX (respectively T_MIN).
+	 * @note when the addition would yield more than T_MAX, it will be T_MAX.
 	 */
 	inline OverflowSafeInt& operator += (const OverflowSafeInt& other)
 	{
-		if ((T_MAX - abs(other.m_value)) < abs(this->m_value) &&
-				(this->m_value < 0) == (other.m_value < 0)) {
-			this->m_value = (this->m_value < 0) ? T_MIN : T_MAX ;
+		if ((this->m_value > 0) && (other.m_value > 0) && (T_MAX - other.m_value) < this->m_value) {
```

Comment:
lots of unneeded `()`. Not sure that is a good or bad thing, just an observation at this point :D